lib/glific_web/providers/gupshup/controllers/message_event_controller.ex

```diff
@@ -54,6 +54,8 @@ defmodule GlificWeb.Providers.Gupshup.Controllers.MessageEventController do
   defp update_status(conn, params, status) do
     bsp_message_id = get_in(params, ["payload", "gsId"]) || get_in(params, ["payload", "id"])
     Communications.Message.update_bsp_status(bsp_message_id, status, params)
+    # we are making an additional query to db to fetch message for sending message status subscription
+    Communications.Message.fetch_and_publish_message_status(bsp_message_id)
```

Comment:
Let's move this to communication/message.ex file 
in the  Communications.Message.update_bsp_status/3

So that it works for all the bsp